src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkCustomCommandLine.java

```diff
@@ -754,10 +761,13 @@ Builder addFormatted(Object object, String format) {
       return this;
     }
 
-    CommandLine build(boolean flagPerLine) {
+    CommandLine build(boolean flagPerLine, @Nullable RepositoryMapping mainRepoMapping) {
       if (arguments.isEmpty()) {
         return CommandLine.empty();
       }
+      if (mainRepoMapping != null) {
+        arguments.add(mainRepoMapping);
```

Comment:
This makes the builder not reusable. This can cause an issue in the following case:

```
args = ctx.actions.args()
args.add(ctx.label)
ctx.actions.run(..., arguments = [args])
ctx.actions.run(..., arguments = [args])
```

`build()` will get called twice and you end up with multiple repo mappings. Can you reproduce this in a test?

A solution is: if the repo mapping is not null, drain the args to an array with an extra slot and then add the repo mapping to the end of the array.